[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39826 [quant][graphmode] Quantizing traced modules
* #39825 [quant][graphmode] Remove dedup pass
* **#39645 [quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing**

Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

jerryzh168 committed:
[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing

Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1868,11 +1867,12 @@ def forward(self, x):
-                       .check_not("aten::relu") \
-                       .check_not("aten::relu_") \
-                       .run(model.graph)
+            for tracing in [True, False]:
```
This test: test_qbatch_norm_relu can only run in FBGEMM as batch norm is not supported as an op in qnnpack, please add a check for qengine for this test.

jerryzh168 replied:
OK, I'll do skipIfNoFBGEMM

jerryzh168 committed:
Update on "[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing"


Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing"


Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing"


Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing"


Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing"


Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing"


Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing"


Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

[ghstack-poisoned]
